src/main/java/teammates/logic/core/LogsService.java

```diff
@@ -0,0 +1,183 @@
+package teammates.logic.core;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Logging.EntryListOption;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Severity;
+
+import teammates.common.exception.LogsServiceException;
+import teammates.common.util.Config;
+
+/**
+ * Logs service that leverages on Google Cloud Logging to create and search for logs.
+ * This service is not supported on the development server.
+ *
+ * @see <a href="https://cloud.google.com/logging/docs">https://cloud.google.com/logging/docs</a>
+ */
+public class LogsService {
```

Comment:
There is already a `GoogleCloudLoggingService` we can use from master - can merge with that

Reply:
Yep, rebased this branch to take in these newer changes from master. Can work with those instead
Most parts have now been shifted to the newer `GoogleCloudLoggingService` and `LogsProcessor` files